src/CommunityToolkit/Xamarin.CommunityToolkit/Converters/EnumToIntConverter.shared.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Globalization;
+using Xamarin.CommunityToolkit.Extensions.Internals;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Converters
+{
+	/// <summary>
+	///     Converts an <see cref="Enum" /> to its underlying <see cref="int" /> value.
+	/// </summary>
+	public class EnumToIntConverter : ValueConverterExtension, IValueConverter
+	{
+		/// <summary>
+		/// Convert a default <see cref="Enum"/> (i.e., extending <see cref="int"/>) to corresponding underlying <see cref="int"/>
+		/// </summary>
+		/// <param name="value"><see cref="Enum"/> value to convert</param>
+		/// <param name="targetType">Unused: type of the binding target property</param>
+		/// <param name="parameter">Unused: Additional parameter for converter</param>
+		/// <param name="culture">Unused: Culture to use in the converter</param>
+		/// <returns>The underlying <see cref="int"/> value of the passed enum value</returns>
+		/// <exception cref="ArgumentException">If value is not an enumeration type</exception>
+		public object Convert(object? value, Type? targetType, object? parameter, CultureInfo? culture) =>
+			value is Enum ? System.Convert.ToInt32(value) : throw new ArgumentException($"{value?.GetType().Name} is not a valid enumeration type");
```

Comment:
```csharp
public enum Name : long
{
    Option
}
```

Just want to higlight that enum can be extended from **long** (Not sure how many people use it though)

Reply:
I think the <summary> of Convert clearly states that a default enum is expected. I would strongly assume that a long-based enum will work as well, but I don't think anyone really uses it anyway